src/main/java/com/puppycrawl/tools/checkstyle/checks/design/VisibilityModifierCheck.java

```diff
@@ -154,6 +154,38 @@
  * &lt;module name=&quot;VisibilityModifier&quot;/&gt;
  * </pre>
  * <p>
+ * Example with default values:
+ * </p>
+ * <pre>
+ * public class MyClass {
+ * private int myPrivateField1;
+ *
+ * int field1;               // violation, must have a visibility modifier
+ *
+ * protected String field2;  // violation, protected visibility is not allowed
+ *
+ * public int field3 = 42;   // violation, not static final, immutable, matching the pattern
+ * public long serialVersionUID = 1L;
+ * public static final int field4 = 42;
+ *
+ * public final int field5 = 42;          // violation, public immutable fields are allowed
+ * public final java.lang.String notes;   // violation, public immutable fields are not allowed
+ *
+ * public final Set&lt;String&gt; mySet1 = new HashSet&lt;&gt;();  // violation, HashSet is mutable
+ * public final ImmutableSet&lt;String&gt; mySet2;           // violation, mutable
+ * public final ImmutableMap&lt;String, Object&gt; objects1; // violation, mutable
```

Comment:
`// violation, immutable type not in config`

Reply:
Done.